scripts/create-release-tag.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+set -eu
+
+err () {
+    echo "$@" >&2
+}
+
+# Tag format: v1.2.3 OR v1.2.3-<tag>
+check_is_valid_github_tag () {
+    if ! [[ "$1" =~ ^v[0-9]+\.[0-9]+\.[0-9]+(-[a-z0-9]+)?$ ]]; then
+        err "Invalid tag: $1"
+        err "Tag must be in the form of vX.Y.Z or vX.Y.Z-<tag>"
+        exit 2
+    fi
+}
+
+# Check branch is main or beta, only allow tags without -<tag> on main
+check_current_branch () {
+  git branch --show-current | grep -E "^(main|beta)$" || {
```

Comment:
What I predict is that people will `git push -f` their experimental branch to `beta` and then publish from beta, meaning that it won't solve the issue.

It seems that beta tags are used by the community, whether it's expected or not.

I would propose the following, as another step / process to setup (out of scope for this PR as it does not currently downgrade the experience)

`beta` & `latest` tags are officially supported (or maybe just `latest`), we use `alpha`, `experimental`, `test` or some kid of other tag for our own testing purposes, we make this very clear in our github documentation: `latest`, `beta` are officially supported, anything else is not, use at your own risk. 

If we want greater protection, we could also ensure that these `alpha`-like versions are privately published, meaning that we don't risk community users accidentally falling on a version which we might not want used.

`beta` (beta branch) is ahead of `latest` (main branch) and `beta` cannot be overridden with a `git push -f`, only incremented, the same way that we deal with `main`. This will ensure that versions don't have unexpectedly missing features.

This process:
- allows freedom to safely publish experimental versions
- communicates what we officially support
- ensures stability for officially supported versions

What do you think?

Reply:
Created a ticket for this: https://github.com/liveblocks/liveblocks.io/issues/937

Reply:
Agreed, happy to consider both `main` and `beta` as special branches, so that:
- we can only publish `beta` tags from a `beta` branch (and they _must_ be named vX.Y.Z-betaN)
- we can only publish `latest` tags from the `main` branch (and they _must_ be named vX.Y.Z)

But other than that, it's fine to publish other NPM tags freely from any branch, no need for extra conventions or limits. We'll probably only ever need `test` for that, but hey, any tag name is fine AFAIAC. This way, we avoid multiple people from using `test` at the same time, too.

If I've used these custom tags in the past, I've always (manually) removed them after I was done testing things in NPM anyway. Just to keep our view of https://www.npmjs.com/package/@liveblocks/client?activeTab=versions clean. (Ironically, I just noticed there still is a `test` version out there now—I'll clean that up right away 😉)

Sounds good?

Reply:
> I've always (manually) removed them after I was done testing things in NPM anyway.

Done so by running the following, btw:

```
npm dist-tag rm "@liveblocks/node" <tagname>
npm dist-tag rm "@liveblocks/zustand" <tagname>
npm dist-tag rm "@liveblocks/redux" <tagname>
npm dist-tag rm "@liveblocks/react" <tagname>
npm dist-tag rm "@liveblocks/client" <tagname>
npm dist-tag rm "@liveblocks/core" <tagname>
```

Reply:
Done, added the checks in the `npm run publish` script

Reply:
Great! I like your propositions. Let's put that in place in a subsequent PR